lib/network/modules/components/Edge.js

```diff
@@ -676,17 +685,29 @@ class Edge {
       else {
         // Ignore the orientations.
         this.labelModule.pointToSelf = true;
-        var x, y;
-        var radius = this.options.selfReferenceSize;
-        if (node1.shape.width > node1.shape.height) {
-          x = node1.x + node1.shape.width * 0.5;
-          y = node1.y - radius;
-        }
-        else {
-          x = node1.x + radius;
-          y = node1.y - node1.shape.height * 0.5;
-        }
-        point = this._pointOnCircle(x, y, radius, 0.125);
+        var angle = 0.125 * 2 * Math.PI;
+        if (
+          typeof this.options.selfReference.angle !== "undefined" &&
+          this.options.selfReference.angle !== null
```

Comment:
This is not important but `null == undefined` so `this.options.selfReference.angle != null` would achieve the same result.